Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1669 @@
+    { 
+        private TypeInfoWrapper _typeInfo;
+        private ComTypes.VARDESC _varDesc;
+        private IntPtr _varDescPtr;
+        private string _name;
+
+        public string Name { get => _name; }
+
+        public TypeInfoVar(TypeInfoWrapper typeInfo, int index)
+        {
+            _typeInfo = typeInfo;
+
+            ((ComTypes.ITypeInfo)_typeInfo).GetVarDesc(index, out _varDescPtr);
+            _varDesc = StructHelper.ReadStructure<ComTypes.VARDESC>(_varDescPtr);
+            
+            int _cNames = 0;
+            var _names = new string[1];
+            if (_varDesc.memid != (int)TypeLibConsts.MEMBERID_NIL)
+            {
+                _cNames = 0;
+                ((ComTypes.ITypeInfo)_typeInfo).GetNames(_varDesc.memid, _names, _names.Length, out _cNames);
+                _name = _names[0];
+            }
+            else
+            {
+                _name = "{unknown}";     // VBA Constants appear in the typelib with no name
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_varDescPtr != IntPtr.Zero) ((ComTypes.ITypeInfo)_typeInfo).ReleaseVarDesc(_varDescPtr);
+            _varDescPtr = IntPtr.Zero;
+        }
+
+        public void Document(StringLineBuilder output) 
+        {
+            output.AppendLine("- field: " + _name + " [id 0x" + _varDesc.memid.ToString("X") + "]");
+        }
+    }
+    
+    /// <summary>
+    /// A base class for exposing an enumerable collection through an index based accessor
+    /// </summary>
+    /// <typeparam name="TItem">the collection element type</typeparam>
+    public class IIndexedCollection<TItem> : IEnumerable<TItem>
+        where TItem : class
+    {
+        IEnumerator IEnumerable.GetEnumerator() => new IIndexedCollectionEnumerator<IIndexedCollection<TItem>, TItem>(this);
+        public IEnumerator<TItem> GetEnumerator() => new IIndexedCollectionEnumerator<IIndexedCollection<TItem>, TItem>(this);
+
+        virtual public int Count { get => throw new NotImplementedException(); }
+        virtual public TItem GetItemByIndex(int index) => throw new NotImplementedException();
+    }
+
+    /// <summary>
+    /// The enumerator implementation for IIndexedCollection
+    /// </summary>
+    /// <typeparam name="TCollection">the IIndexedCollection<> type</typeparam>
+    /// <typeparam name="TItem">the collection element type</typeparam>
+    public class IIndexedCollectionEnumerator<TCollection, TItem> : IEnumerator<TItem>, IDisposable
+        where TCollection : IIndexedCollection<TItem>
+        where TItem : class
+    {
+        private TCollection _collection;
+        private int _collectionCount;
+        private int _index = -1;
+        TItem _current;
+
+        public IIndexedCollectionEnumerator(TCollection collection)
+        {
+            _collection = collection;
+            _collectionCount = _collection.Count;       
+        }
+
+        public void Dispose()
```

Comment:
Hmm. I don't see any other need for the `IDisposable` for this class. Are we sure it still needs that interface? Otherwise we can remove that.

Reply:
sure